packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -942,15 +942,6 @@ export function performWorkOnRoot(
         markRootSuspended(root, lanes, NoLane, didAttemptEntireTree);
       }
       break;
-    } else if (exitStatus === RootDidNotComplete) {
-      if (enableProfilerTimer && enableComponentPerformanceTrack) {
-        finalizeRender(lanes, now());
-      }
-      // The render unwound without completing the tree. This happens in special
-      // cases where need to exit the current render without producing a
-      // consistent tree or committing.
-      const didAttemptEntireTree = !workInProgressRootDidSkipSuspendedSiblings;
-      markRootSuspended(root, lanes, NoLane, didAttemptEntireTree);
```

Comment:
In the `else` branch we just fall through to `finishConcurrentRender` since we won't be concurrent and it doesn't match the explicit exitStatus checks.

Therefore we enter the same path as RootSuspendedWithDelay and we can use the same code path.